### PR TITLE
Order confirmation flow with download of images

### LINKED ISSUE #57: Stripe modal payment confirmation
Replace confirmation view with a modal window

**MUST HAVE**

* [ ] Modal window with confirmation text
* [ ] redirect to root

### LINKED ISSUE #60: Mail payment / order confirmation
Mail confirmation of order with either

* attached images(s)
* download link(s)

**MUST HAVE**

* [ ] send mail with one of the above

### PR DESCRIPTION
**Pull request by ocardinaux**
**Needs a `bundle install` and a `db:seed`**

* order with zipfile download
* Adding factories and tests to cart, cart_cat

Closes #57 
Closes #60

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,13 +10,15 @@ gem 'aws-sdk-s3', require: false
...
-gem 'dotenv-rails'
+gem 'dotenv-rails', '~> 2.7.2'
 gem 'faker', '~> 1.8', '>= 1.8.7'
 gem 'jbuilder', '~> 2.5'
 gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 3.11'
 gem 'rails', '~> 5.2.3'
+gem 'rubyzip', '~> 1.2.2'
 gem 'sass-rails', '~> 5.0'
+gem 'sentry-raven', '~> 2.9.0'
```
Comment (ocardinaux) [inline Gemfile]: This is for a plugin on heroku that alerts on production problems

File: Gemfile [hunk 2]
```diff
@@ -10,13 +10,15 @@ gem 'aws-sdk-s3', require: false
...
 gem 'coffee-rails', '~> 4.2'
 gem 'devise', '~> 4.6.0'
-gem 'dotenv-rails'
+gem 'dotenv-rails', '~> 2.7.2'
 gem 'faker', '~> 1.8', '>= 1.8.7'
 gem 'jbuilder', '~> 2.5'
 gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 3.11'
 gem 'rails', '~> 5.2.3'
+gem 'rubyzip', '~> 1.2.2'
```
Comment (ocardinaux) [inline Gemfile]: This is user to zip the files of an ordered cart

File: Gemfile [hunk 3]
```diff
@@ -28,9 +30,11 @@ end
 
 group :development do
   gem 'annotate', '~> 2.7', '>= 2.7.4'
+  gem 'awesome_print'
```
Comment (ocardinaux) [inline Gemfile]: Both gems for debugging

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,4 +1,12 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  before_action :set_raven_context
+
+  private
+
+  def set_raven_context
```
Comment (ocardinaux) [inline app/controllers/application_controller.rb]: For heroku plugin

File: db/seeds.rb [hunk 5]
```diff
@@ -34,3 +43,23 @@
   dl_cat_image = URI.parse(Faker::LoremFlickr.image).open
   cat.item_picture.attach(io: dl_cat_image, filename: 'cat_image.jpg', content_type: "image/jpeg")
 end
+
+(1..5).to_a.each do |i|
+  puts "Creating closed (ordered) Cart #{i}"
```
Comment (ocardinaux) [inline db/seeds.rb]: Add ordered carts to seed

File: app/models/cart.rb [hunk 6]
```diff
@@ -35,4 +35,8 @@ def cart_user
   def total_cart_price
     cart_cats.map(&:total_cat_price).sum
   end
+
+  def zipfile
+    ZipCartContent.perform(cart: id)
```
Comment (TheMindset) [inline app/models/cart.rb]: J'ai pas compris cette ligne de commande ?

File: app/controllers/carts_controller.rb [hunk 7]
```diff
@@ -13,4 +13,10 @@ def destroy
     current_user.set_current_cart
     redirect_to root_path
   end
+
+  def download_cart_zip
```
Comment (TheMindset) [inline app/controllers/carts_controller.rb]: J'ai compris ce que fait exactement cete méthode ?

File: spec/models/cart_spec.rb [hunk 8]
```diff
@@ -23,30 +23,78 @@
...
+        expect(ordered_cart.order_placed).to be_instance_of ActiveSupport::TimeWithZone
+      end
+    end
+  end
+
+  describe '#cart_user' do
+    it { expect(cart.cart_user).to be_instance_of(User) }
+  end
+
+  describe '#total_cart_price' do
```
Comment (denispasin) [inline spec/models/cart_spec.rb]: let(:cat1) { create(:cat) }

File: app/assets/javascripts/application.js [hunk 9]
```diff
@@ -12,7 +12,7 @@
 //
 //= require rails-ujs
 //= require activestorage
-//= require turbolinks
+//# require turbolinks
```
Comment (ocardinaux) [inline app/assets/javascripts/application.js]: Removed this, and life is better! 😄

Comment (denispasin) [reply app/assets/javascripts/application.js]: C'est dommage, c'est bien turbolink

**Pull request merged**